Remove requested store and forward messages

neonknight64 opened the pull request:
## Description
- These changes remove the store and forward messages from a local node after a destination node has requested these messages and the response message has been queued or sent.
- Added remove_message function to store and forward db.
- Added RemoveMessages store and forward request and provided implementations for the StoreAndForwardRequester and StoreAndForwardMock handle_request.

## Motivation and Context
These changes will remove old store and forward messages from the db after it has been provided or queued to be sent to the destination node.

## How Has This Been Tested?
Added a remove_messages test for the store and forward db.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

sdbondi commented on comms/dht/src/store_forward/database/mod.rs:
```diff
@@ -57,6 +57,17 @@ impl StoreAndForwardDatabase {
             .await
     }
 
+    pub async fn remove_message(&self, message_id: i32) -> Result<usize, StorageError> {
+        self.connection
+            .with_connection_async(move |conn| {
+                diesel::delete(stored_messages::table)
+                    .filter(stored_messages::id.eq(message_id))
+                    .execute(conn)
+                    .map_err(Into::into)
+            })
+            .await
+    }
+
```
```suggestion
    pub async fn remove_messages(&self, message_ids: Vec<i32>) -> Result<usize, StorageError> {
        self.connection
            .with_connection_async(move |conn| {
                diesel::delete(stored_messages::table)
                    .filter(stored_messages::id.eq_any(message_ids))
                    .execute(conn)
                    .map_err(Into::into)
            })
            .await
    }

```

neonknight64 replied:
Great! Change applied.

sdbondi reviewed (commented):
Looks good - Rather use [`eq_any`](http://docs.diesel.rs/diesel/expression_methods/trait.ExpressionMethods.html#method.eq_any) and pass in the vec of ids to delete them in one query

Commit:
- These changes will remove old store and forward messages from the db after it has been provided or queued to be sent to the destination node.
- These changes remove the store and forward messages from a local node after a destination node has requested these messages and the response message has been queued or sent.
- Added remove_message function to store and forward db.
- Added RemoveMessages store and forward request and provided implementations for the StoreAndForwardRequester and StoreAndForwardMock handle_request.
- Added a remove_messages test for the store and forward db.

sdbondi reviewed (approved):
Looks good!
